File: 2024/php/day01_2.php
<?php

$numbers = file_get_contents('inputs/day01.txt');
$file = fopen("inputs/day01.txt", "r");
if ($file) {
    $sum = [];

    while (($line = fgets($file)) !== false) {
        $numbers = explode(' ', $line);
        $leftGroup[] = $numbers[0];
        $rightGroup[] = $numbers[3];
    }
    
    foreach ($leftGroup as $leftNumber) {
        $count = 0;
        foreach ($rightGroup as $rightNumber) {
            if ($leftNumber == $rightNumber) {
                $count++;
            }
        }
        if ($count > 0) {
            $sum[] = $leftNumber * $count;
        }
    }

    print_r("The sum is: " . array_sum($sum));
    fclose($file);
}

File: 2024/go/day01_2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("inputs/day01.txt")
	if err != nil {
		fmt.Println("couldn't read the file")
	}
	leftGroup := []int{}
	rightGroup := []int{}
	sum := []int{}

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()

		splitString := strings.Split(line, " ")

		leftNumber, _ := strconv.Atoi(splitString[0])
		leftGroup = append(leftGroup, leftNumber)

		rightNumber, _ := strconv.Atoi(splitString[3])
		rightGroup = append(rightGroup, rightNumber)
	}

	for _, leftNumber := range leftGroup {
		count := 0
		for _, rightNumber := range rightGroup {
			if leftNumber == rightNumber {
				count++
			}
		}
		if count > 0 {
			sum = append(sum, leftNumber*count)
		}
	}

	sumOfArray(sum)
}

func sumOfArray(sum []int) {
	result := 0

	for i := 0; i < len(sum); i++ {
		result += sum[i]
	}
	println(result)
}
